WooCommerce/Classes/ViewModels/Order Details/Shipping Labels/BoldableTextView.swift

```diff
@@ -0,0 +1,113 @@
+import SwiftUI
+
+/// A text view that takes in a string whose substrings within two asterisks are bolded.
+///
+/// For example, initializing with a string "*Never* will I *stop learning*." will bold the substrings "Never" and "stop learning".
+///
+struct BoldableTextView: View {
+    fileprivate struct Element {
+        let content: String
+        let isBold: Bool
+
+        init(content: String, isBold: Bool) {
+            if isBold {
+                self.content = content.replacingOccurrences(of: boldRegionMark, with: "")
+            } else {
+                self.content = content
+            }
+            self.isBold = isBold
+        }
+    }
+
+    fileprivate static let boldRegionMark = "*"
+
+    private let elements: [Element]
+
+    init(_ content: String) {
+        elements = content.parseBoldableElements()
+    }
+
+    var body: some View {
+        elements.map { $0.toTextView }
+            .reduce(into: Text(""), { result, text in
+                result = result + text
+            })
+    }
+}
+
+private extension BoldableTextView.Element {
```

Comment:
Maybe Element could / should live in a view model/presenter? That way this logic, plus the little bit of logic above that maps elements and then reduces plus the string manipulation could be part of the view model as well? 

That way, in theory, the view could be completely free of logic. What do you think?

Reply:
The element model is now a separate struct `BoldableElement` in the same file as `BoldableTextParser`! feel free to comment on anything on the latest code 🙂